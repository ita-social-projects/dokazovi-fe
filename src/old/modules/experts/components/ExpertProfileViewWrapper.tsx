import React, { useState, useEffect, useCallback } from 'react';
import { useHistory, useParams } from 'react-router-dom';
import { IExpert } from '../../../lib/types';
import { getExpertById } from '../../../lib/utilities/API/api';
import ExpertProfileView from './ExpertProfileView';
import { setGALocation } from '../../../../utilities/setGALocation';
import { ERROR_404 } from '../../../lib/constants/errors';

const ExpertProfileViewWrapper: React.FC = () => {
  const { expertId } = useParams<{ expertId: string }>();
  const history = useHistory();
  const [loadedExpert, setLoadedExpert] = useState<IExpert>();
  const [statusCode, setStatusCode] = useState<number>();

  const fetchExpert = useCallback(async () => {
    try {
      const expertResponse = await getExpertById(Number(expertId));
      setLoadedExpert(expertResponse.data);
    } catch (error) {
      setStatusCode(404);
    }
  }, [expertId]);

  useEffect(() => {
    setGALocation(window);
  }, []);

  useEffect(() => {
    fetchExpert();
  }, [fetchExpert]);

  if (statusCode === 404) {
    history.push(ERROR_404);
  }

  return <>{loadedExpert && <ExpertProfileView expert={loadedExpert} />}</>;
};

export default ExpertProfileViewWrapper;
