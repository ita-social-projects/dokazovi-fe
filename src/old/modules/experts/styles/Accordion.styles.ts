import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(() => ({
  fullName: {
    marginBottom: '20px',
  },
  contacts: {
    width: '91%',
    fontSize: '14px',
    lineHeight: '14px',
    wordWrap: 'break-word',
  },
  links: {
    color: '#2051FF',
  },
  accordionWrapper: {
    marginBottom: '50px',
  },
  accordion: {
    width: '100%',
    backgroundColor: '#E5E5E5',
  },
  accordionSummary: {
    padding: 0,
    userSelect: 'none',
    '&:hover': {
      cursor: 'pointer',
    },
  },
  accordionSummaryTitle: {
    display: 'flex',
  },
  accordionDetails: {
    marginTop: '25px',
  },
  accordionDetailsItem: {
    marginTop: '15px',
  },
  showContacts: {
    fontSize: '18px',
    lineHeight: '18px',
    fontWeight: 700,
    fontFamily: 'Raleway',
    width: '91%',
  },
  expandIcon: {
    width: '20px',
    height: '20px',
    fontSize: '36px',
    lineHeight: '10px',
    fontWeight: 700,
    margin: '0px',
    padding: '0px',
  },
  contactIcons: {
    marginRight: '7px',
  },
  headingsDivider: {
    height: '4px',
    background: '#000000',
    margin: '2px 0 0 0',
  },
}));
