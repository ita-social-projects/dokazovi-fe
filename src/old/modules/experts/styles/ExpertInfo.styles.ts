import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(() => ({
  personalInfo: {
    marginBottom: '20px',
  },
  divider: {
    marginBottom: '20px',
  },
  avatar: {
    margin: '0 20px 20px 0',
    width: '240px',
    height: '240px',
  },
  directionList: {
    '& *': {
      marginRight: '4px',
    },
  },
}));
