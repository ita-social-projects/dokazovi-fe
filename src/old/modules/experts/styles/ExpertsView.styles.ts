import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(
  () => ({
    container: {
      display: 'flex',
      flexWrap: 'wrap',
      margin: '7px 0px 15px 0px',
      justifyContent: 'flex-start',
      alignItems: 'center',
    },
    title: {
      width: '100%',
      margin: '0 0 28px 15px',
      fontSize: '28px',
      lineHeight: '28px',
      fontWeight: 'bold',
    },
    selectedFilters: {
      margin: '0px 15px 15px 0px',
      fontFamily: 'Raleway',
      fontStyle: 'normal',
      fontWeight: 500,
      fontSize: '14px',
      lineHeight: '1.4',
    },
    totalFilters: {
      margin: '0px 15px 15px 0px',
      fontFamily: 'Literata',
      fontStyle: 'italic',
      fontWeight: 'normal',
      fontSize: '14px',
      lineHeight: '1.29',
    },
    divider: {
      margin: '0px 15px 15px 0px',
      fontFamily: 'Raleway',
      fontStyle: 'normal',
      fontWeight: 500,
      fontSize: '16px',
      lineHeight: '1.4',
    },
  }),
  {
    name: 'ExpertsView',
  },
);
