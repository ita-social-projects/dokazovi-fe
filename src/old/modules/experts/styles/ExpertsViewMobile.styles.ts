import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(
  () => ({
    container: {
      padding: 'auto',
      minHeight: 'calc(100vh - 353px)',
    },
    headerContainer: {
      boxSizing: 'border-box',
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      margin: '25px 10px 20px',
    },
    divider: {
      width: '40%',
      height: '1px',
      background: 'black',
    },
    header: {
      lineHeight: '22px',
      fontSize: '24px',
      fontFamily: 'Raleway',
      fontWeight: 700,
      paddingInline: '15px',
      whiteSpace: 'nowrap',
    },
    showAllButton: {
      margin: '20px 0 30px 0',
      width: '217px',
      height: '48px',
      border: 'none',
      borderRadius: '24px',
      fontSize: '17px',
      backgroundColor: '#4FDFFF',
    },
  }),
  {
    name: 'ExpertsViewMobile',
  },
);
