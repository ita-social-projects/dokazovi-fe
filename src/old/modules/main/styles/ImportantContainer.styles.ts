import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(
  {
    container: {
      minHeight: 455,
    },
    containerMobile: {
      minHeight: 585,
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
    },
  },
  { name: 'ImportantContainer' },
);
