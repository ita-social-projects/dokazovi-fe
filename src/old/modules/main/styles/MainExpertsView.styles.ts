import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(() => ({
  experts: {
    display: 'grid',
    width: '70rem',
    height: '18rem',
    gridTemplateColumns: '4fr 2fr 4fr 1fr 2fr 4fr',
    gridTemplateRows: '1fr 1fr 1fr 1fr',
    columnGap: '1rem',
    rowGap: '1rem',
    margin: '0 auto',
  },
  container: {
    minHeight: '400px',
    position: 'relative',
  },
  loading: {
    position: 'absolute',
    top: 'calc(50% - 30px)',
    msTransform: 'translateY(-50%)',
    transform: 'translateY(-50%),',
  },
  item_0: {
    gridArea: '1 / 1 / 5 / 2',
  },
  item_1: {
    gridArea: '1 / 2 / 3 / 3',
  },
  item_2: {
    gridArea: '3 / 2 / 5 / 3',
  },
  item_3: {
    gridArea: '1 / 3 / 5 / 4',
  },
  item_4: {
    gridArea: '1 / 4 / 2 / 5',
  },
  item_5: {
    gridArea: '2 / 4 / 3 / 5',
  },
  item_6: {
    gridArea: '3 / 4 / 4 / 5',
  },
  item_7: {
    gridArea: '4 / 4 / 5 / 5',
  },
  item_8: {
    gridArea: '1 / 5 / 3 / 6',
  },
  item_9: {
    gridArea: '3 / 5 / 5 / 6',
  },
  item_10: {
    gridArea: '1 / 6 / 5 / 7',
  },
}));
