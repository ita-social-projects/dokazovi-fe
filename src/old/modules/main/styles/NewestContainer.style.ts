import makeStyles from '@material-ui/core/styles/makeStyles';

export const useStyles = makeStyles(
  () => ({
    container: {
      minHeight: 600,
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
    },
  }),
  { name: 'NewestContainer' },
);
