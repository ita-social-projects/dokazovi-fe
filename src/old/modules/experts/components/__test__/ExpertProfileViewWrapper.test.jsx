/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { combineReducers, configureStore, createSlice } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import {
  mockExpertMaterials,
  mockProperties,
  mockMaterials,
} from './mocks/expertProfileWrapper';
import ExpertProfileViewWrapper from '../ExpertProfileViewWrapper';
import { getExpertById } from '../../../../lib/utilities/API/api';
import { ThemeProvider } from '@material-ui/core';
import { MAIN_THEME } from '../../../../../styles/theme';

const mockedStore = {
  expertMaterials: mockExpertMaterials,
  properties: mockProperties,
  materials: mockMaterials,
};

const mockExpertMaterialsSlice = createSlice({
  name: 'mockExpertMaterials',
  initialState: mockExpertMaterials,
  reducers: {},
});

const mockPropertiesSlice = createSlice({
  name: 'mockExpertMaterials',
  initialState: mockProperties,
  reducers: {},
});

const mockMaterialsSlice = createSlice({
  name: 'mockExpertMaterials',
  initialState: mockMaterials,
  reducers: {},
});

const mockStr = configureStore({
  reducer: combineReducers({
    expertMaterials: mockExpertMaterialsSlice.reducer,
    properties: mockPropertiesSlice.reducer,
    materials: mockMaterialsSlice.reducer,
  }),
});

jest.mock('../../../../../models/store', () => ({
  store: {
    getState: () => mockedStore,
  },
}));

jest.mock('../../../../lib/utilities/API/api', () => ({
  getExpertById: jest.fn(),

  getActivePostTypes: () =>
    Promise.resolve({
      data: [{ id: 1, name: 'Стаття' }],
    }),
  getActiveDirections: () =>
    Promise.resolve({
      data: [
        {
          id: 3,
          name: 'surgery',
          label: 'Хірургія',
          color: '#7aebbf',
          hasDoctors: true,
          hasPosts: true,
        },
      ],
    }),
}));

jest.mock('react-ga');

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
  useParams: () => ({ expertId: '10' }),
  useLocation: () => ({
    pathname: '/test',
  }),
}));

describe('ExpertProfileViewWrapper component tests', () => {
  it('should render ExpertProfileViewWrapper component', async () => {
    const { asFragment } = render(
      <Provider store={mockStr}>
        <MemoryRouter>
          <ExpertProfileViewWrapper />
        </MemoryRouter>
      </Provider>,
    );

    await waitFor(() => expect(asFragment()).toMatchSnapshot());
  });

  it('should render expert profile', async () => {
    getExpertById.mockImplementation(() =>
      Promise.resolve({
        data: {
          id: 10,
          firstName: 'Марія',
          lastName: 'Марієнко',
          bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
          qualification: 'Лікар вищої категорії',
          phone: '+380956456969',
          email: 'masha@mail.com',
          socialNetworks: ['facebook/link'],
        },
      }),
    );

    render(
      <ThemeProvider theme={MAIN_THEME}>
        <Provider store={mockStr}>
          <MemoryRouter>
            <ExpertProfileViewWrapper />
          </MemoryRouter>
        </Provider>
      </ThemeProvider>,
    );

    await waitFor(() =>
      expect(screen.getByTestId('expert-info')).toBeInTheDocument(),
    );
  });

  it('should render error 404', async () => {
    getExpertById.mockImplementation(() => Promise.reject());

    render(
      <Provider store={mockStr}>
        <MemoryRouter>
          <ExpertProfileViewWrapper />
        </MemoryRouter>
      </Provider>,
    );

    await waitFor(() =>
      expect(mockHistoryPush).toHaveBeenCalledWith(`/error_404`),
    );
  });
});
