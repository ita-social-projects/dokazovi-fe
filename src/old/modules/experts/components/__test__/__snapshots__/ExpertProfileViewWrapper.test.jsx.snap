// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpertProfileViewWrapper component tests should render ExpertProfileViewWrapper component 1`] = `<DocumentFragment />`;
